analyzers/dart-analyze/utils/extractor.py

```diff
@@ -0,0 +1,190 @@
+                        self.consume()
+                        continue
+
+                    # Handle code blocks
+                    if "{% prettify dart tag=pre+code %}" in line:
+                        in_code_block = True
+                        line = "```dart"
+                    elif "{% endprettify %}" in line:
+                        in_code_block = False
+                        line = "```"
+
+                    if in_code_block:
+                        # If the line we're extracting is part of a code block
+                        # then we need to remove '[!' & '!]' from the line. This
+                        # syntax is used to for showing squiggly lines in the docs.
+                        # We don't support this hence we remove it.
+                        line = line.replace("[!", "").replace("!]", "")
+
+                    description += f"{line}\n"
+                    self.consume()
+                description = description.strip()
+
+                issues.append(
+                    {"code": code, "title": title, "description": description}
+                )
+            else:
+                self.consume()
+
+        return issues
+
+
+class IssueExtractor:
+    """Extracts dart's diagnostic and linter rules."""
+
+    LINTER_RULES_URL = (
+        "https://github.com/dart-lang/site-www/raw/main/src/_data/linter_rules.json"
+    )
+    DIAGNOSTIC_RULES_URL = "https://raw.githubusercontent.com/dart-lang/site-www/main/src/tools/diagnostic-messages.md"
+    GROUP_CATEGORY_MAP = {
+        "style": "antipattern",
+        "errors": "bug-risk",
+        "pub": "antipattern",
+    }
+
+    def __init__(self) -> None:
+        """
+        Extracts dart analyze issues.
+
+        Args:
+            directory (str): The directory to use to clone the dart \
+                analyze repository
+        """
+        self._issues = []
+
+    @property
+    def issues(self) -> List[Issue]:
+        """The list of issues extracted from the dart analyze repository."""
+        if len(self._issues) > 0:
+            return self._issues
+
+        self._issues = self.fetch_rules()
+        return self._issues
+
+    def fetch_rules(self) -> List[Issue]:
+        """Fetches all dart analyze rules."""
+        return [*self.fetch_linter_rules(), *self.fetch_diagnostic_rules()]
+
+    def fetch_diagnostic_rules(self) -> List[Issue]:
+        """Fetches the list of diagnostic rules"""
+        response = urlopen(self.DIAGNOSTIC_RULES_URL)  # skipcq: BAN-B310
+        if response.code != 200:
+            raise Exception(f"Failed to fetch diagnostic rules {response.code}")
+
+        rules = DiagnosticRulesParser(response.read().decode()).get_issues()
+        return [Issue(**rule, category="bug-risk") for rule in rules]
+
+    def fetch_linter_rules(self) -> List[Issue]:
+        """Fetches the list of linter rules."""
+        response = urlopen(self.LINTER_RULES_URL)  # skipcq: BAN-B310
+        if response.code != 200:
```

Comment:
```suggestion
        if not response.ok:
```

Reply:
`ok` is not an attribute of the response object.